zebrad/src/application.rs

```diff
@@ -235,11 +235,13 @@ impl Application for ZebradApp {
         let mut components = self.framework_components(command)?;
 
         // Load config *after* framework components so that we can
-        // report an error to the terminal if it occurs.
+        // report an error to the terminal if it occurs (unless used with the 'generate' cmd).
         let config = match command.config_path() {
-            Some(path) => match self.load_config(&path) {
-                Ok(config) => config,
-                Err(e) => {
+            Some(path) => match (&command.cmd, self.load_config(&path)) {
+                (_, Ok(config)) => config,
+                // Ignore errors loading the config when generating a default config.
+                (&Some(ZebradCmd::Generate(_)), Err(_e)) => Default::default(),
```

Comment:
Optional: The list of commands could be implemented as a method on `ZebradCmd` instead, so that we are more likely to update it when we add new commands.

Reply:
Good idea, applied in https://github.com/ZcashFoundation/zebra/pull/7014/commits/35f253f53a9872e99beab0ae72cfee6bcf1b8323.